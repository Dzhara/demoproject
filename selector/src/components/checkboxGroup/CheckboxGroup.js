import React from "react";
import PropTypes from "prop-types";
import Checkbox from "../checkbox/Checkbox";
import style from "./CheckboxGroup.module.scss";

export default class CheckboxGroup extends React.Component {

  render() {
    return (
      <div className={style.checkboxGroupWrapper}>
        {this.props.source.map((option) => {
          const optionLabel =
            option.label !== undefined && option.label !== null
              ? option.label
              : option;
          const optionValue =
            option.value !== undefined && option.value !== null
              ? option.value
              : option;
          const checked = this.props.selectedItems.includes(optionValue);
          return (<div className={style.checkboxGroupItem} key={`checkbox-item-${optionValue}`}>
            <Checkbox
              value={optionValue}
              label={optionLabel}
              checked={checked}
              onChange={state => {
                this.props.onSelected(optionValue, state);
              }}
            /></div>)
        })}
      </div>
    );
  }
}

CheckboxGroup.propTypes = {
  source: PropTypes.arrayOf(Object).isRequired,
  selectedItems: PropTypes.array
}

CheckboxGroup.defaultProps = {
  source: undefined
}
