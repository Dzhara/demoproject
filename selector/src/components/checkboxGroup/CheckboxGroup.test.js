import React from "react";
import renderer from "react-test-renderer";
import CheckboxGroup from "./CheckboxGroup";

describe("CheckboxGroup", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <CheckboxGroup
        source={[{ label: "test", value: "test" }]}
        selectedItems={[]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
