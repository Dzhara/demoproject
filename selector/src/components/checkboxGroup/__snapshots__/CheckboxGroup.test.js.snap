// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup snapshot renders 1`] = `
<div
  className="checkboxGroupWrapper"
>
  <div
    className="checkboxGroupItem"
  >
    <div
      className="ui checkbox badge "
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={null}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label>
        test
      </label>
    </div>
  </div>
</div>
`;
